vllm/entrypoints/openai/serving_chat.py

```diff
@@ -49,23 +57,46 @@ async def create_chat_completion(
             return self.create_error_response(
                 "logit_bias is not currently supported")
 
-        try:
-            prompt = self.tokenizer.apply_chat_template(
-                conversation=request.messages,
-                tokenize=False,
-                add_generation_prompt=request.add_generation_prompt)
-        except Exception as e:
-            logger.error(
-                f"Error in applying chat template from request: {str(e)}")
-            return self.create_error_response(str(e))
+        if "chatglm3" in request.model:
+            [*history, query] = request.messages
+            query = query["content"]
+            prompt_ids = (self.tokenizer.build_chat_input(
+                query=query, history=history,
+                role="user").input_ids.squeeze().tolist())
+            prompt = self.tokenizer.decode(prompt_ids)
+            try:
+                token_ids = self._validate_prompt_and_tokenize(
+                    request, prompt_ids=prompt_ids)
+            except ValueError as e:
+                return self.create_error_response(str(e))
+        elif "Baichuan" in request.model:
+            prompt_ids = self.tokenizer.build_chat_input(request.messages)
+            prompt = self.tokenizer.decode(prompt_ids)
```

Comment:
is this needed? we have the prompt ids already

Reply:
This question is consistent with the one below